src/Disks/S3/DiskS3.h

```diff
@@ -173,7 +181,7 @@ class DiskS3 final : public IDiskRemote
     inline static const String RESTORE_FILE_NAME = "restore";
 
     /// Key has format: ../../r{revision}-{operation}
-    const re2::RE2 key_regexp {".*/r(\\d+)-(\\w+)$"};
+    const re2::RE2 key_regexp {".*/r(\\d+)(-[\\w\\d\\-\\.]+)?-(\\w+)$"};
```

Comment:
We need not check that this part is a correct hostname. It's just a mark to make different object names for different replicas. It may be some random UUID or something like an "replica number", but I use hostname, because it's already exists and looks potentially useful for some purposes like debugging in future.